website/content/docs/concepts/username-templating.mdx

```diff
@@ -0,0 +1,143 @@
+---
+layout: docs
+page_title: Username Templating
+sidebar_title: Username Templating
+description: >-
+  Username templating are used in some secret engines to allow operators to define
+  how dynamic usernames are generated.
+---
+
+# Username Templating
+
+Some of the secrets engines that generate dynamic users for external systems provide the ability for Vault operators
+to customize how usernames are generated for said external systems. This uses the
+[Go template language](https://golang.org/pkg/text/template/). This page describes the basics of using these templates
+for username generation but does not go into great depth of using the templating language for more advanced usages.
+See the API documentation for the given secret engine to determine if it supports username templating and for more
+details on using it with that engine.
+
+~> When customizing how usernames are generated, take care to ensure you have enough randomness to ensure uniqueness
+   otherwise multiple calls to create the credentials may interfere with each other.
+
+In addition to the functionality built into the Go template language, a number of additional functions are available:
+
+## Available Functions
+
+### String/Character Manipulation
+
+`lowercase` - Lowercases the provided value.<br/>
+**Example**: `{{.FieldName | lowercase}}`
+
+`replace` - Find/replace on the provided value.<br/>
+**Example**: `{{.FieldName | replace - _}}`
+
+`truncate` - truncates the previous value to the provided number of characters.<br/>
+**Example**: `{{.FieldName | truncate 10}}`
+
+`truncate_sha256` - Truncates the previous value to the provided number of characters. The last 8 characters of the
+truncated value will be the first 8 characters of the SHA256 hash of the truncated characters.<br/>
```

Comment:
```suggestion
new value will be replaced by the first 8 characters of the SHA256 hash of the truncated characters.<br/>
```

Assuming I understand the operation, I think this reads more clearly. Let me know if I do not understand correctly, which is entirely possibly.

Reply:
That looks good and matches with what it does. This function is a bit complicated to explain so anything that can be done to make this clearer is appreciated.

Reply:
It reads clearer to me now, thanks!